tests/unit/lib/test_opensearch_data.py

```diff
@@ -0,0 +1,174 @@
+
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_put_get_set_object_implementation_specific_behavior(self, scope):
+        """Test putting and getting objects in/from the secret store."""
+        self.secret_store.put_object(scope, "key-obj", {"name1": "val1"}, merge=True)
+        self.secret_store.put_object(
+            scope, "key-obj", {"name1": None, "name2": "val2"}, merge=True
+        )
+        self.assertDictEqual(
+            self.secret_store.get_object(scope, "key-obj"), {"name1": None, "name2": "val2"}
+        )
+
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_delete(self, scope):
+        """Test delete key."""
+        self.secret_store.delete(scope, "nonexistent")
+        self.assertIsNone(self.secret_store.get(scope, "nonexistent"))
+
+        self.secret_store.put(scope, "key1", "val1")
+        self.secret_store.delete(scope, "key1")
+        self.assertIsNone(self.secret_store.get(scope, "key1"))
+
+        self.secret_store.put_object(scope, "key-obj", {"key1": "val1", "key2": "val2"})
+        self.secret_store.delete(scope, "key-obj")
+        self.assertFalse(self.secret_store.has(scope, "key-obj"))
+        self.assertIsNone(self.secret_store.get(scope, "key-obj"))
+
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_nullify_value(self, scope):
+        """Test definging a key as `None`."""
+        self.secret_store.put(scope, "key1", "val1")
+        self.secret_store.put(scope, "key1", None)
+        self.assertFalse(self.secret_store.has(scope, "key1"))
+
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_nullify_obj(self, scope):
+        """Test iteratively filling up an object with `None` values."""
+        self.secret_store.put_object(scope, "key-obj", {"key1": "val1", "key2": "val2"})
+        self.secret_store.put_object(scope, "key-obj", {"key1": None, "key2": "val2"}, merge=True)
+        self.secret_store.put_object(scope, "key-obj", {"key2": None}, merge=True)
+        self.assertTrue(self.secret_store.has(scope, "key-obj"))
+        secret = self.secret_store.get_object(scope, "key-obj")
+        self.assertIsNone(secret.get("key1"))
+        self.assertIsNone(secret.get("key2"))
+
+
+@pytest.mark.usefixtures("only_with_juju_secrets")
+class TestHelperSecrets(TestHelperDatabag):
+    def setUp(self) -> None:
+        super().setUp()
+        self.secret_store = self.charm.secrets
+
+    @override
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_put_get_set_object_implementation_specific_behavior(self, scope):
+        """Test putting and getting objects in/from the secret store."""
+        self.secret_store.put_object(scope, "key-obj", {"name1": "val1"}, merge=True)
+        self.secret_store.put_object(
+            scope, "key-obj", {"name1": None, "name2": "val2"}, merge=True
+        )
+        self.assertDictEqual(self.secret_store.get_object(scope, "key-obj"), {"name2": "val2"})
```

Comment:
This difference shouldn't matter, as normally we use `get()` to access dictionary keys.

(TODO: I'll try to double-check the code for sure, but based on my current knowledge + tests we should be already safe.)

Reply:
Specific commit added where all direct references are checked with `get()` before